src/stepFunctions/commands/visualizeStateMachine.ts

```diff
@@ -29,257 +29,339 @@ export interface messageObject {
+            existingVisualization.showPanel()
 
-        return setupWebviewPanel(documentUri, textDocument)
-    } catch (err) {
-        vscode.window.showInformationMessage(
-            localize(
-                'AWS.stepfunctions.visualisation.errors.rendering',
-                'There was an error rendering State Machine Graph, check logs for details.'
-            )
-        )
+            return existingVisualization.getPanel()
+        }
 
-        logger.debug('Unable to setup webview panel.')
-        logger.error(err as Error)
-    }
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await cache.updateCache(globalStorage)
 
-    return
-}
+            return this.createNewVisualization(textDocument)
+        } catch (err) {
+            vscode.window.showInformationMessage(
+                localize(
+                    'AWS.stepfunctions.visualisation.errors.rendering',
+                    'There was an error rendering State Machine Graph, check logs for details.'
+                )
+            )
 
-async function setupWebviewPanel(
-    documentUri: vscode.Uri,
-    textDocument: vscode.TextDocument
-): Promise<vscode.WebviewPanel> {
-    const logger: Logger = getLogger()
-    let isPanelDisposed = false
-
-    // Create and show panel
-    const panel = vscode.window.createWebviewPanel(
-        'stateMachineVisualization',
-        makeWebviewTitle(documentUri),
-        {
-            preserveFocus: true,
-            viewColumn: vscode.ViewColumn.Beside,
-        },
-        {
-            enableScripts: true,
-            localResourceRoots: [
-                ext.visualizationResourcePaths.localWebviewScriptsPath,
-                ext.visualizationResourcePaths.visualizationLibraryCachePath,
-                ext.visualizationResourcePaths.stateMachineCustomThemePath,
-            ],
-            retainContextWhenHidden: true,
+            logger.debug('Unable to setup webview panel.')
+            logger.error(err as Error)
         }
-    )
 
-    async function sendUpdateMessage(updatedTextDocument: vscode.TextDocument) {
-        const isValid = await isDocumentValid(updatedTextDocument)
+        return
+    }
 
-        if (isPanelDisposed || !panel.webview) {
-            return
-        }
+    public deleteVisualization(visualizationToDelete: AslVisualization) {
+        this.managedVisualizations.delete(visualizationToDelete)
+    }
 
-        logger.debug('Sending update message to webview.')
+    private createNewVisualization(textDocument: vscode.TextDocument): vscode.WebviewPanel | void {
+        const newVisualization = new AslVisualization(textDocument)
+        this.managedVisualizations.add(newVisualization)
 
-        panel.webview.postMessage({
-            command: 'update',
-            stateMachineData: updatedTextDocument.getText(),
-            isValid,
+        newVisualization.getOnVisualizationDisposeEvent()(() => {
```

Comment:
Registering an event returns a disposable. Add this to `ext.subscriptions` so that it gets cleaned up on extension deactivation.